src/components/scenes/SendScene2.tsx

```diff
@@ -0,0 +1,924 @@
+      return (
+        <WarningCard
+          title={s.strings.warning_scam_title}
+          points={[
+            s.strings.warning_scam_message_financial_advice,
+            s.strings.warning_scam_message_irreversibility,
+            s.strings.warning_scam_message_unknown_recipients
+          ]}
+          footer={s.strings.warning_scam_footer}
+          marginRem={[1.5, 1]}
+        />
+      )
+    }
+    return null
+  }
+
+  const recordFioObtData = async (spendTarget: EdgeSpendTarget, currencyCode: string, txid: string) => {
+    const { nativeAmount, publicAddress: payeePublicAddress = '', otherParams = {} } = spendTarget
+    const { fioAddress: payeeFioAddress } = otherParams
+    if (fioSender != null) {
+      const { fioAddress: payerFioAddress, fioWallet, memo, skipRecord = false } = fioSender
+      if (payeeFioAddress != null && payerFioAddress != null && fioWallet != null) {
+        // if (guiMakeSpendInfo.fioPendingRequest != null) {
+        // const { fioPendingRequest: pendingRequest } = guiMakeSpendInfo
+        // try {
+        //   await recordSend(fioWallet, fioAddress, {
+        //     fioRequestId: pendingRequest.fio_request_id,
+        //     payeeFioAddress: pendingRequest.payee_fio_address,
+        //     payerPublicAddress: pendingRequest.payer_fio_public_key,
+        //     payeePublicAddress: pendingRequest.content.payee_public_address,
+        //     amount: pendingRequest.content.amount,
+        //     currencyCode: pendingRequest.content.token_code.toUpperCase(),
+        //     chainCode: pendingRequest.content.chain_code.toUpperCase(),
+        //     txid: edgeSignedTransaction.txid,
+        //     memo
+        //   })
+        // } catch (e: any) {
+        //   const message = e?.message ?? ''
+        //   message.includes(FIO_FEE_EXCEEDS_SUPPLIED_MAXIMUM) ? showError(s.strings.fio_fee_exceeds_supplied_maximum_record_obt_data) : showError(e)
+        // }
+        // } else if ((guiMakeSpendInfo.publicAddress != null || publicAddress != null) && (!skipRecord || edgeSignedTransaction.currencyCode === FIO_STR)) {
+        if (!skipRecord) {
+          const { publicAddress: payerPublicAddress } = await coreWallet.getReceiveAddress()
+          const amount = nativeAmount ?? '0'
+          const chainCode = coreWallet.currencyInfo.currencyCode
+
+          try {
+            recordSend(fioWallet, payerFioAddress, {
+              payeeFioAddress,
+              payerPublicAddress,
+              payeePublicAddress,
+              amount: amount && div(amount, cryptoExchangeDenomination.multiplier, DECIMAL_PRECISION),
+              currencyCode: currencyCode,
+              chainCode,
+              txid,
+              memo
+            })
+          } catch (e: any) {
+            showError(e)
+          }
+        }
+      }
+    }
+  }
+
+  const handleSliderComplete = useHandler(async (resetSlider: () => void) => {
+    // TODO:
+    // 1. FIO functionality
```

Comment:
It looks like the FIO stuff is implemented.

Reply:
Not completely. Paying a FIO request is still not supported. We're using the old SendScene for that.